### PR TITLE
[Utils] add utilities for checking if certain utilities are properly documented

### PR DESCRIPTION
**Pull request by sayakpaul**
# What does this PR do?

We have a bunch of attention processors, normalization layers, and activation layers in the codebase and they have thin documentation. This is fine because users can always see their usage in the context of a model or a pipeline to get the fuller picture. 

But since we document them in separate pages too, (such as [this](https://huggingface.co/docs/diffusers/main/en/api/attnprocessor), [this](https://huggingface.co/docs/diffusers/main/en/api/activations), and [this](https://huggingface.co/docs/diffusers/main/en/api/normalization)), I think it makes sense to ensure these docs properly reflect what all we support. 

This PR adds utilities to do that. @stevhliu I think you will like it :-)

### CODE REVIEW

**Comment by sayakpaul** [comment]
@stevhliu any idea why the doc build fails?

**Comment by stevhliu** [comment]
I'm not sure, the formatting all looks correct to me but for reason `autodoc.py` can't find them! cc @mishig25 have you encountered this before?

**Comment by sayakpaul** [comment]
@mishig25 a gentle ping.

**Comment by mishig25** [comment]
let me check

File: docs/source/en/api/normalization.md [hunk 1]
```diff
@@ -29,3 +29,19 @@ Customized normalization layers for supporting various models in 🤗 Diffusers.
...
 
 [[autodoc]] models.normalization.AdaGroupNorm
+
+## AdaLayerNormContinuous
+
+[[autodoc]] models.normalization.AdaLayerNormContinuous
+
+## LayerNorm
+
+[[autodoc]] models.normalization.LayerNorm
```
Comment (mishig25) [inline docs/source/en/api/normalization.md:39]: `LayerNorm` is the reason why doc-builder is not building.

I've inspected a bit. 
All other norms are diffuser defined classes:
```
>>> diffusers.models.normalization.GlobalResponseNorm
<class 'diffusers.models.normalization.GlobalResponseNorm'>
```
whereas `LayerNorm` seems to be alias to pytorch defined class:
```
>>> diffusers.models.normalization.LayerNorm
<class 'torch.nn.modules.normalization.LayerNorm'>
```

Therefore, as a simple fix: I'd suggest replacing `[[autodoc]] models.normalization.LayerNorm` with something like: `You can also use LayerNorm and add markdown link to pytorch layernorm doc`

Comment (sayakpaul) [reply docs/source/en/api/normalization.md:39]: Thanks @mishig25. I think it'd be okay to remove LayerNorm from our doc because our implementation is really a special case and is already supported in the latest versions of PyTorch.

**Comment by sayakpaul** [comment]
@stevhliu @yiyixuxu this is ready for review.

File: src/diffusers/models/normalization.py [hunk 2]
```diff
@@ -151,6 +151,18 @@ def forward(self, x: torch.Tensor, emb: torch.Tensor) -> torch.Tensor:
 
 
 class AdaLayerNormContinuous(nn.Module):
+    r"""
+    Adaptive normalization layer with a norm layer (layer_norm or rms_norm).
+
+    Args:
+        embedding_dim (`int`): Embedding dimension to use during projection.
+        conditioning_embedding_dim (`int`): Dimension of the input condition.
+        elementwise_affine (`bool`): Boolean flag to denote if affine transformation should be applied.
```
Comment (stevhliu) [inline src/diffusers/models/normalization.py:160]:
```suggestion
        elementwise_affine (`bool`, defaults to `True`): Boolean flag to denote if affine transformation should be applied.
```

File: src/diffusers/models/normalization.py [hunk 3]
```diff
@@ -151,6 +151,18 @@ def forward(self, x: torch.Tensor, emb: torch.Tensor) -> torch.Tensor:
...
+        eps (`int`): Epsilon factor.
```
Comment (stevhliu) [inline src/diffusers/models/normalization.py:161]:
```suggestion
        eps (`float`, defaults to 1e-5): Epsilon factor.
```

File: src/diffusers/models/normalization.py [hunk 4]
```diff
@@ -151,6 +151,18 @@ def forward(self, x: torch.Tensor, emb: torch.Tensor) -> torch.Tensor:
...
+        bias (`bias`): Boolean flag to denote if bias should be use.
```
Comment (stevhliu) [inline src/diffusers/models/normalization.py:162]:
```suggestion
        bias (`bias`, defaults to `True`): Boolean flag to denote if bias should be use.
```

File: src/diffusers/models/normalization.py [hunk 5]
```diff
@@ -151,6 +151,18 @@ def forward(self, x: torch.Tensor, emb: torch.Tensor) -> torch.Tensor:
...
+        norm_type (`str`): Normalization layer to use. Values supported: "layer_norm", "rms_norm".
```
Comment (stevhliu) [inline src/diffusers/models/normalization.py:163]:
```suggestion
        norm_type (`str`, defaults to `"layer_norm"`): Normalization layer to use. Values supported: "layer_norm", "rms_norm".
```

File: src/diffusers/models/normalization.py [hunk 6]
```diff
@@ -188,6 +200,16 @@ def forward(self, x: torch.Tensor, conditioning_embedding: torch.Tensor) -> torc
     # Has optional bias parameter compared to torch layer norm
     # TODO: replace with torch layernorm once min required torch version >= 2.1
     class LayerNorm(nn.Module):
+        r"""
+        LayerNorm with the bias parameter.
+
+        Args:
+            dim (`int`): Dimensionality to use for the parameters.
+            eps (`int`): Epsilon factor.
```
Comment (stevhliu) [inline src/diffusers/models/normalization.py:208]:
```suggestion
            eps (`float`, defaults to 1e-5): Epsilon factor.
```

File: src/diffusers/models/normalization.py [hunk 7]
```diff
@@ -188,6 +200,16 @@ def forward(self, x: torch.Tensor, conditioning_embedding: torch.Tensor) -> torc
...
+            elementwise_affine (`bool`): Boolean flag to denote if affine transformation should be applied.
```
Comment (stevhliu) [inline src/diffusers/models/normalization.py:209]:
```suggestion
            elementwise_affine (`bool`, defaults to `True`): Boolean flag to denote if affine transformation should be applied.
```

File: src/diffusers/models/normalization.py [hunk 8]
```diff
@@ -188,6 +200,16 @@ def forward(self, x: torch.Tensor, conditioning_embedding: torch.Tensor) -> torc
...
+            bias (`bias`): Boolean flag to denote if bias should be use.
```
Comment (stevhliu) [inline src/diffusers/models/normalization.py:210]:
```suggestion
            bias (`bias`, defaults to `True`): Boolean flag to denote if bias should be use.
```

File: src/diffusers/models/normalization.py [hunk 9]
```diff
@@ -210,6 +232,15 @@ def forward(self, input):
 
 
 class RMSNorm(nn.Module):
+    r"""
+    RMS Norm as introduced in https://arxiv.org/abs/1910.07467 by Zhang et al.
+
+    Args:
+        dim (`int`): Number of dimensions to use for `weights`. Only effective when `elementwise_affine` is True.
+        eps (`float`): Small value to use when calculating the reciprocal of the square-root.
+        elementwise_affine (`bool`): Boolean flag to denote if affine transformation should be applied.
```
Comment (stevhliu) [inline src/diffusers/models/normalization.py:241]:
```suggestion
        elementwise_affine (`bool`, defaults to `True`): Boolean flag to denote if affine transformation should be applied.
```

**Review by stevhliu — APPROVED** [review]
LGTM, thanks!

**Comment by sayakpaul** [comment]
Gentle ping @yiyixuxu.

**Comment by yiyixuxu** [comment]
can you take a look here @DN6?

**Review by yiyixuxu — APPROVED** [review]
super cooll! thanks!

File: docs/source/en/api/normalization.md [hunk 10]
```diff
@@ -29,3 +29,43 @@ Customized normalization layers for supporting various models in 🤗 Diffusers.
...
+[[autodoc]] models.normalization.LpNorm
+
+## CogView3PlusAdaLayerNormZeroTextImage
+[[autodoc]] models.normalization.CogView3PlusAdaLayerNormZeroTextImage
+
+## CogVideoXLayerNormZero
+[[autodoc]] models.normalization.CogVideoXLayerNormZero
+
+## MochiRMSNormZero
+[[autodoc]] models.normalization.MochiRMSNormZero
```
Comment (hlky) [inline docs/source/en/api/normalization.md:68]:
```suggestion
[[autodoc]] models.transformers.transformer_mochi.MochiRMSNormZero
```

**Review by hlky — APPROVED** [review]
Nice!
